### ISSUE TITLE
Come up with final project name

### ISSUE THREAD

**Issue by lucasjkr** [lucasjkr/opencommerce#83]
I hate thinking of names but I need to come up with one.

**Comment by arnisjuraga** [comment]
"CoponaX" 🗡

**Comment by lucasjkr** [comment]
Wish we could be on a combined project! But we have different goals and ideas, seemed much easier to go on a new path :)

**Comment by zemguru** [comment]
:DD

**Comment by lucasjkr** [comment]
Here's where I differ:

1 - I think shared hosts for ecommerce websites are a really BAD idea. They worked in the 90's, but its 2017. VPS's are cheap as heck and much more secure. I don't think any concessions should be made to make it run on a shared environment. That should be totally unsupported, and if it DOES happen to work on a shared host, then we say "best of luck!"

2 - Likewise, given that case, there is no reason to target PHP5.x for anything. It's end of life. Only getting maintenance releases for a year. Again, shared hosts use it, so that's what Copona wants to support. 

3 - OC has unsupported databases in its code (MSSQL, Postgres), which Copona inherited. Further, it has two database access methods for Mysql, either mysql or MPDO. By saying MySQL/PDO is the only acceptable method, I've gotten to rewrite 90% of the models in my project to use prepared statements with bound parameters. Which are safer. Plus the queries become easier to read.

     SELECT * FROM table WHERE column = :column
versus
    SELECT * FROM table WHERE column = $this->db->escape($column)

Add a few more criteria to the query and joins, and Opencarts queries become difficult to decipher. By needing to support the lowest common denominator of what COULD be available to a shard host, we lose out on better methods.

4 - Back to shared hosts - security wise, I think its flawed to leave anything more than necessary in the web accessible directories. OC3 wised up to this and now advises users to move the system directory. Copona wants to run from the public_html directory only, leaving access to files that should be protected only a typo in htaccess away.

I guess it comes down to, while I'm NOT a security guru, I still think there are reasonable steps any shopping cart software should take.

Opencart is successful with a huge installed base, so they can't easily take those steps that could benefit the users because of compatibility concerns. I get that. But one of the the bonus' of starting from a cleaner slate is not having to worry about that installed base. Just need to make known your system requirements and (eventually) make tools to help migrate them over.

**Comment by arnisjuraga** [comment]
That's not so bad :)

>1 - I think shared hosts for ecommerce websites are a really BAD idea. They worked in the 90's, but its 2017. VPS's are cheap as heck and much more secure. I don't think any concessions should be made to make it run on a shared environment. That should be totally unsupported, and if it DOES happen to work on a shared host, then we say "best of luck!"
Yes. This is the thing, which I am still taking advocating. Yes, you benefit from VPS. But it depends on project model. 
If we are creating the system, which is structured, extendable and configurable, then, I believe, it's not a problem to make it compatible with different platforms - Windows/Linux VPS and Shared including. 

If I have a configuration, involving Sphinx search - it's just impossible to run on shared hostings.

So, in Copona case, I believe in base model "for everyone" scalable "for anyone";

> 2 - Likewise, given that case, there is no reason to target PHP5.x for anything. It's end of life. Only getting maintenance releases for a year. Again, shared hosts use it, so that's what Copona wants to support.

IF there are a reason, to use PHP 7 intentionally, I would wote for that. At the moment - there are much to do, and this question can be easily postponed, and everyone will come to it anyway. I just don't believe it's  a critical issue at the moment, because yes, PHP 7 is not yet so popular. But may be at the moment you or Copona makes a sable release, things will be much different.

**Comment by arnisjuraga** [comment]
> 3 - OC has unsupported databases in its code (MSSQL, Postgres), 
Nope, this is something we just have not finished, but no - there is no need for anything else as only MySQL support and PDO at the moment. This we have agreed already.

> 4 - Back to shared hosts - security wise, 
 The same, as 1st: if the system is scalable and compatible for different platforms, there should ne not a problem to make it run from subfolder in any moment.
Copona have already configurable "storage/private" variable, and everything else should use relative paths. 

What I mean with everything said here - I believe we are on the same path, and of course, as you know, it's a bit harder to get to the decision in team :), but for growing projects it should become a team one way or another :)

**Comment by lucasjkr** [comment]
First Note: - The project has a name "LibreCommerce"  (only because someone took LibreCart already, but abandoned it. Lame!)

I'll write a new issue and direct you and anyone else to look over there soon.

**Comment by lucasjkr** [comment]
**1 - Shared Hosts**
I can only say, I don't think any compromise should be made for shared hosts. If it was up to me, I would actively take steps to prevent from running on a shared host, but I can't do that.

Again, by default on Shared hosts, users have read access to everyone elses directory. So, you can read the database config, you can connect to the database, you can read every credential (email, ftp, credit card API's, everything!). And since you have the DB password and are connecting from the same host, you can also write new API keys to the database (for payment processing options whose keys are stored in oc_settings, or somewhere else), so a misbehaving user could redirect payments to themselves.

The options for mischief are unlimited, unless the host is acting proactively and running things in a non-standard way.

And you or your client can get a VPS unshared (within the VM) for $5 / month. They don't want to administer a server, that's fine. We should concentrate on tools to make it so they don't need to worry about things, NOT make concessions to let them run in a share environment.

So that is my biggest gripe, something I can not support or endorse! :)

**2 - PHP Version**
We always encounter pushback about upping the minimum PHP version to 7.0, as you cite that 5.6 is still technically supported. I can't point to anything specific yet, but again, I'd rather not even think about it - code the way we want to code, and if it runs under 5.6, fine, but don't claim to support it :) 

**3 - Database**
Last I heard, we went in a circle and you still wanted MySQLi as well as PDO, which prevented parameterized queries.

It's ok, we can continue to collaborate and share :)
